test/integration/image_test.go

```diff
@@ -0,0 +1,132 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestImageBuild(t *testing.T) {
+	type validateFunc func(context.Context, *testing.T, string)
+	profile := UniqueProfileName("image")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(15))
+	defer Cleanup(t, profile, cancel)
+
+	// Serial tests
+	t.Run("serial", func(t *testing.T) {
+		tests := []struct {
+			name      string
+			validator validateFunc
+		}{
+			{"NormalBuild", validateNormalImageBuild},
+			{"BuildWithBuildArg", validateImageBuildWithBuildArg},
+			{"BuildWithDockerIgnore", validateImageBuildWithDockerIgnore},
+			{"BuildWithSpecifiedDockefile", validateNormalImageBuildWithSpecifiedDockefile},
+		}
+		for _, tc := range tests {
+			if ctx.Err() == context.DeadlineExceeded {
+				t.Fatalf("Unable to run more tests (deadline exceeded)")
+			}
+			if t.Failed() {
+				t.Fatalf("Previous test failed, not running dependent tests")
+			}
+
+			t.Run(tc.name, func(t *testing.T) {
+				tc.validator(ctx, t, profile)
+				if t.Failed() && *postMortemLogs {
+					PostMortemLogs(t, profile)
+				}
+			})
+		}
+	})
+}
+
+// normal test case for minikube image build, with -t parameter
+func validateNormalImageBuild(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "./testdata/image-build/test-normal", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Fatalf("failed to start minikube with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// normal test case for minikube image build, with -t and -f parameter
+func validateNormalImageBuildWithSpecifiedDockefile(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "-f", "inner/Dockerfile", "./testdata/image-build/test-f", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	data, _ := io.ReadAll(rr.Stdout)
+
+	output := string(data)
+	fmt.Println(output)
+	if err != nil {
+		t.Fatalf("failed to start minikube with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// test case for bug in https://github.com/kubernetes/minikube/issues/12384
+// build with --build-opt
+func validateImageBuildWithBuildArg(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "--build-opt=build-arg=ENV_A=aaasdftggdf", "--build-opt=no-cache", "./testdata/image-build/test-arg", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	data, _ := io.ReadAll(rr.Stdout)
+
+	output := string(data)
```

Comment:
This can be simplified:
```suggestion
	output := rr.Stdout.String()
```